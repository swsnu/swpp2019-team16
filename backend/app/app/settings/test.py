from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
