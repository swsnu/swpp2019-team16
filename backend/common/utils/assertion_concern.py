class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
