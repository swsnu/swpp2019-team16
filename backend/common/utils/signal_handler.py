import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
