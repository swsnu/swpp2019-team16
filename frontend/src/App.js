import React from 'react';
import { Route, Switch } from 'react-router-dom';
import IntroPage from './pages/IntroPage/IntroPage';
import RequestPage from './pages/RequestPage/RequestPage';
import RequestCallPage from 'pages/RequestCallPage/RequestCallPage';
import WaitingPage from './pages/WaitingPage/WaitingPage';
import GroupPage from './pages/GroupPage/GroupPage';
import DetailPage from './pages/DetailPage/DetailPage';
import FinalPage from './pages/FinalPage/FinalPage';
import RegisterPage from './pages/RegisterPage/RegisterPage';
import LoginPage from './pages/LoginPage/LoginPage';
import DriverDetailPage from './pages/DriverDetailPage/DriverDetailPage';
import RiderDetailPage from './pages/RiderDetailPage/RiderDetailPage';
import DriverFinalPage from './pages/DriverFinalPage/DriverFinalPage';
import RiderFinalPage from './pages/RiderFinalPage/RiderFinalPage';
import NotFoundPage from './pages/NotFoundPage/NotFoundPage';

function App() {
  return (
    <div>
      <Switch>
        <Route component={IntroPage} path="/" exact />
        <Route component={RegisterPage} path="/register" exact />
        <Route component={RequestPage} path="/request" exact />
        <Route component={RequestCallPage} path="/requestcall" exact />
        <Route component={WaitingPage} path="/waiting" exact />
        <Route component={GroupPage} path="/group" exact />
        <Route component={DetailPage} path="/detail" exact />
        <Route component={DriverDetailPage} path="/driverdetail" exact />
        <Route component={DriverFinalPage} path="/driverfinal" exact />
        <Route component={RiderDetailPage} path="/riderdetail" exact />
        <Route component={RiderFinalPage} path="/riderfinal" exact />
        <Route component={FinalPage} path="/final" exact />
        <Route component={LoginPage} path="/login" exact />
        <Route component={NotFoundPage} />
      </Switch>
    </div>
  );
}

export default App;
