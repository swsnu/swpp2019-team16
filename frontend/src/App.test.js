import React from 'react';
import App from './App';
import { MemoryRouter, Route } from 'react-router-dom';
import { renderWithRedux } from './test/utils';

jest.mock('@fullpage/react-fullpage', () => ({ children }) => (
  <div>{children}</div>
));

describe('<App />', () => {
  const state = {
    auth: {
      auth: {
        id: 1,
      },
    },
    user: {
      user: null,
    },
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter>
        <Route component={App} path="/" />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
