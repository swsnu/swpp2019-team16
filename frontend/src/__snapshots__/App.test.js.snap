// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> SHOULD match with snapshot 1`] = `
<div>
  <div>
    <div
      class="sc-bwzfXH kulSxP"
    >
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <h6
              class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            >
              Ya-Ta!
            </h6>
          </div>
        </header>
      </div>
      <main
        class="sc-bdVaJa ekJyAd"
      >
        <div
          class="Introduction"
        >
          <p
            class="MuiTypography-root MuiTypography-h1 MuiTypography-paragraph"
          >
            Welcome to YA-TA!
          </p>
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            What is yata?
          </h3>
          <p
            class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph"
          >
            Ya-Ta helps students who need to urgently get to class. Within ten minutes, Ya-Ta groups students with these similar, urgent, needs, and grabs a taxi for them.
          </p>
          <div
            class="startButton"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Start!
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
